node/src/main/scala/coop/rchain/node/web/ReportingRoutes.scala

```diff
@@ -46,7 +48,15 @@ object ReportingRoutes {
 
     HttpRoutes.of[F] {
       case GET -> Root / "trace" :? BlockHashParam(hash) +& ForceReplayParam(forceReplay) =>
-        Ok(transforResult(hash, blockReportAPI.blockReport(hash, forceReplay.getOrElse(false))))
+        Ok(
+          transforResult(
+            hash,
+            blockReportAPI.blockReport(
+              ByteString.copyFrom(Base16.unsafeDecode(hash)),
```

Comment:
ByteString is basically like String. Please create `QueryParamDecoderMatcher[Blake2b256Hash]` with corresponding `QueryParamDecoder`.
In that way we can validate user input and have friendly response error unlike _unsafeDecode_ which should be only used when we know error cannot happen.